### PR TITLE
WIP: RHPAM-4012 add jakarta validation

### PR DESCRIPTION
**Pull request by rgdoliveira**
**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[[link](https://www.example.com)](https://issues.redhat.com/browse/RHPAM-4012)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* https://github.com/kiegroup/kie-soup/pull/243

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -354,6 +354,7 @@
     <!-- The version greater than 1.0.0.GA is not compatible with GWT 2.8.x -->
     <!-- therefore the property is rewritten in that repository parent -->
     <version.javax.validation>2.0.1.Final</version.javax.validation>
+    <version.jakarta.validation>3.0.1</version.jakarta.validation>
```
Comment (Ginxo) [inline pom.xml:357]: @rgdoliveira I would kee this aligned with the new cxf version upgrading, so 2.0.2

Comment (rgdoliveira) [reply pom.xml]: +1, I updated the PR

**Comment by rgdoliveira** [comment]
Jenkins run fdb

**Comment by Ginxo** [comment]
about FDB error, I've being investigating the `ban duplicated classes` error at `appformer/dashbuilder/dashbuilder-shared/dashbuilder-displayer-api` it and
```
[INFO] +- org.kie.soup:kie-soup-dataset-api:jar:7.64.0-SNAPSHOT:compile
[INFO] |  \- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  +- com.google.gwt:gwt-user:jar:2.9.0:test
[INFO] |  |  +- javax.validation:validation-api:jar:1.0.0.GA:test
[INFO] |  |  +- javax.validation:validation-api:jar:sources:1.0.0.GA:test
```

now we have to exclude validation-api from https://github.com/kiegroup/appformer/blob/9f4b0246f4a61a2824369e8a537a9595cdf69810/dashbuilder/dashbuilder-shared/dashbuilder-displayer-api/pom.xml#L44

I'm afraid it won't be the last one

**Comment by rgdoliveira** [comment]
Jenkins run fdb

**Comment by mareknovotny** [comment]
yes, in such cases it is quicker to run locally FDB with limited goal only to enforcer not packaging or installing anything ;) @rgdoliveira @Ginxo

**Pull request closed without merging**